### PR TITLE
Check if texture is valid in material.getTexture

### PR DESCRIPTION
**Pull request by adamnejm**

### CODE REVIEW

**Comment by thegrb93** [comment]
GetTexture used to return error texture if it was missing rather than nil. Is that not the case anymore?

**Comment by adamnejm** [comment]
From my test I concluded that `Material` returns `error` but GetTexture returned `nil`.

**Comment by thegrb93** [comment]
What about if the material is found but not the texture. like GetTexture("$blahblah")

File: lua/starfall/libs_cl/material.lua [hunk 1]
```diff
@@ -346,11 +346,12 @@ end
 --- Gets a texture from a material
 -- @param path The path of the material (don't include .vmt in the path)
 -- @param texture The texture key to get
--- @return The texture's name
+-- @return The texture's name or nil if texture key isn't found
```
Comment (thegrb93) [inline lua/starfall/libs_cl/material.lua:349]: Returns nil if the material isn't found, error texture if the texture isn't found?

**Comment by adamnejm** [comment]
`Material` function returns a `Material` object no matter what, if it's invalid it'll just be the error material.

We can still call `GetTexture` on that. If the texture *key*, ie. `$basetexture` is valid it will just return that `ITexture`, for invalid textures, ie. `$blahblah` it will return `nil`, so this change checks for that

```lua
material.getTexture("blahblah", "$basetexture") --> "error"
material.getTexture("blahblah", "$blahblah") --> nil

material.getTexture("gm_construct/grass_13", "$basetexture") --> gm_construct/grass1
material.getTexture("gm_construct/grass_13", "$blahblah") --> nil
```

File: lua/starfall/libs_cl/material.lua (hunk 1 above)
Comment (adamnejm) [reply lua/starfall/libs_cl/material.lua:349]: If what's passed to `GetTexture`, so the `texture` parameter of this function isn't found on the given material, then it returns nil.

**Comment by thegrb93** [comment]
Ok. Makes sense. Thanks

**Pull request merged**